sdk/identity/Azure.Identity/src/AadIdentityClient.cs

```diff
@@ -176,7 +176,7 @@ private Request CreateClientSecretAuthRequest(string tenantId, string clientId,
 
             request.Uri.Reset(_options.AuthorityHost);
 
-            request.Uri.AppendPath(tenantId);
+            request.Uri.AppendPath(tenantId, true);
```

Comment:
Why is `escape` true in this file and not elsewhere? I don't see anything from the previous code to warrant it, or was it wrong before?

Reply:
Good points. I've changed the default multiple times resulting in inconsistent calls, I'll clean up all your comments.